openml-lightgbm/lightgbm-provider/src/test/java/com/feedzai/openml/provider/lightgbm/LightGBMBinaryClassificationModelTest.java

```diff
@@ -236,4 +249,115 @@ private boolean compareDoubles(final double a, final double b, final int exclude
+     * If there are any differences in the models within such folders they are reported
+     * in a colored report (requires python3 with termcolor installed).
+     *
+     * @param referenceModelsFolder Folder with reference models
+     * @param newModelsFolder       Folder with new models to compare to reference
+     * @return true if there are no differences and all the dependencies are installed, false otherwise.
+     * @throws IOException          In case the process errors.
+     * @throws InterruptedException In case awaiting for the process to finish fails.
+     */
+    private boolean compareModelFilesAndDoPrettyReport(
+            final String referenceModelsFolder,
+            final String newModelsFolder) throws IOException, InterruptedException {
+
+        // Check for differences and report in detailed manner!
+        final Process p = new ProcessBuilder(
+                "python",
+                "diff_models.py",
+                referenceModelsFolder,
+                newModelsFolder
+        ).start();
+
+        // Fetch process output
+        BufferedReader reader = new BufferedReader(new InputStreamReader(p.getInputStream()));
+        StringBuilder builder = new StringBuilder();
+        String line = null;
+        while ((line = reader.readLine()) != null) {
+            builder.append(line);
+            builder.append(System.getProperty("line.separator"));
+        }
+        final String processOutput = builder.toString();
+
+        p.waitFor();
+        logger.error(processOutput);
+
+        return (p.exitValue() == 0);
+    }
+
+    /**
+     * Asserts that the two files have the same content.
+     *
+     * @param name      name of the file to compare for the assert message.
+     * @param filepath1 path to the first file
+     * @param filepath2 path to the second file
+     * @throws IOException Raised in case of failure reading the files
+     */
+    private void assertEqualFileContents(final String name,
+                                         final Path filepath1,
+                                         final Path filepath2) throws IOException {
+
+        File file1 = new File(filepath1.toString());
+        File file2 = new File(filepath2.toString());
+
+        assertThat(FileUtils.contentEquals(file1, file2))
+                .as(String.format("%s file comparison", name))
+                .isTrue();
+    }
+
+    /**
+     * This functional test ensures that LightGBM can read a model file and output one exactly alike the one read in.
+     * This is to ensure the new code to rewrite the model read/write layers is completely functional.
+     * The two reference models were generated initially with LightGBM's v3.0.0 code.
+     * The two generated ones will use the current code in the current locale. There should be no mismatches.
+     * @throws URISyntaxException
+     * @throws ModelLoadingException
+     * @throws IOException
+     * @throws InterruptedException
```

Comment:
@since 1.0.19

Reply:
I don't know what happens with since, but is is supposed to be all lowercase `@since` and all similar comments

Reply:
LOOL it's a username, hover above it xD

Reply:
You're tagging the poor guy many times xD

Reply:
There is some one with name SInCE :man_facepalming: I had to put ` to remove the link